docs/page.js

```diff
@@ -1,3 +1,22 @@
+if ( !window.frameElement ) {
+
+// If the page is not yet displayed as an iframe of the index page (navigation panel/working links),
+// redirect to the index page (using the current URL without extension as the new fragment).
+// If this URL itself has a fragment, append it with a dot (since '#' in an URL fragment is not allowed).
+
+    var hash = window.location.hash.substr( 1 );
+
+    if (hash !== '') {
+
+        hash = '.' + hash;
+
+    }
+
+    window.location.replace( window.location.origin + '/docs/#' + window.location.pathname.slice(6, -5) + hash );
```

Comment:
I think you should be ble to do `window.location.pathname.lastIndexOf('docs/')`.